src/lib/yang/alarms.lua

```diff
@@ -410,6 +410,54 @@ function purge_alarms (args)
    return count
 end
 
+local function alarm_key_matches (k1, k2)
+   if k1.resource and k1.resource ~= k2.resource then
+     return false
+   elseif k1.alarm_type_id and k1.alarm_type_id ~= k2.alarm_type_id then
+     return false
+   elseif k1.alarm_type_qualifier and
+          k1.alarm_type_qualifier ~= k2.alarm_type_qualifier then
+     return false
+   end
+   return true
+end
+
+local function compress_alarm (alarm)
+   assert(alarm.status_change)
+   local latest_status_change = alarm.status_change[#alarm.status_change]
+   alarm.status_change = {latest_status_change}
+end
+
+-- This operation requests the server to compress entries in the
+-- alarm list by removing all but the latest state change for all
+-- alarms.  Conditions in the input are logically ANDed.  If no
+-- input condition is given, all alarms are compressed.
+function compress_alarms (key)
+   local function parse_key (key)
+      local t = {}
+      for each in key:gmatch('([^/]+)') do
+         table.insert(t, each)
+      end
+      return {
+         resource = t[1],
```

Comment:
Could we separate out arguments resource and alarm type id (plus optional qualifier)?  I thought we did that before.